packages/core/src/modules/credentials/protocol/BaseCredentialProtocol.ts

```diff
@@ -275,41 +277,49 @@ export abstract class BaseCredentialProtocol<CFs extends CredentialFormatService
    * Retrieve a credential record by connection id and thread id
    *
    * @param connectionId The connection id
+   * @param role The role of the record, i.e. holder or issuer
    * @param threadId The thread id
+   *
    * @throws {RecordNotFoundError} If no record is found
    * @throws {RecordDuplicateError} If multiple records are found
    * @returns The credential record
    */
-  public getByThreadAndConnectionId(
+  public getByThreadIdConnectionIdAndRole(
```

Comment:
I think it is a bit confusing this method to be called `getByThreadIdConnectionIdAndRole` if it's expecting the role _before_ the connectionId. What about swapping the parameters, renaming it or use an options object like we usually do?

Reply:
Yeah I didn't do an object to not make too many changes. But I think that that would be the correct call (using a bunch of optional parameter causes a bug in this PR earlier and an object would've caught that). 

I do think when it's an object I will rename it to `getByQueryFilter` or something equivalent. This would allow to have more options in the future as well. Also happy to pick another name :).

Reply:
@genaris I updated it to `getByProperties` and `findByProperties`. This'll make it expandable in the future as well. Is that good?

Reply:
Oh, I was just complaining about the order of the parameters, but like this is better of course!